pages/studentPortal/web/index.tsx

```diff
@@ -42,7 +44,7 @@ export default function StudentPortalPage() {
       />
       <div className="grid grid-cols-1 gap-4">
         {units.map((it, index) => (
```

Comment:
Maybe you can check the user role above this line. Show the default units for most students, but if role equals freemium we show a FreemiumCourseComponent.

This may not work but something we should think about together.

Reply:
I have done this but broken it into two components like the original.  Is this too much code duplication or do you think it works?